### ISSUE TITLE
Add a conda installer selector widget

### ISSUE THREAD

**Issue by travishathaway** [conda-incubator/conda-dot-org#81]
## Description

We would like to have a widget on the website to figure out which conda installer is right for them.

## Tasks

- [ ] *tbd*

**Comment by bupd** [comment]
I would like to contribute

**Comment by travishathaway** [comment]
@bupd,

Are you still interested in contributing this feature? If so, I can assign this issue to you. I just updated the description of the issue to be a little more clear.

Let me know if you are still interested or have any questions.

**Comment by travishathaway** [comment]
All,

I have marked this as a https://github.com/conda-incubator/conda-dot-org/labels/good%20first%20issue in case anyone else would like to work on it. @bupd  will have priority because he asked first. If he does not want to work on this issue, I will assign it to the second person who comments.

**Comment by Shreyas-SAS** [comment]
@travishathaway if @bupd is not available, I can work on this issue. Please assign it to me in that case. Thanks.

**Comment by bhavukkalra** [comment]
@Shreyas-SAS @travishathaway Any progress on this?
I see that this issue isn't assigned yet. Do let me know, If someone gets assigned to this.
Maybe we can collaborate on it!

**Comment by travishathaway** [comment]
@Shreyas-SAS,

You've already been assigned to another issue. To more fairly distribute our good first issues, I will be assigning this to @bhavukkalra instead.

@bupd, I'm going assume you're no longer interested given it's been a couple of days since I messaged you.

**Comment by Shreyas-SAS** [comment]
@travishathaway yeah, it's fine. I will also start my work on the other issue in an hour or so. Thanks.

**Comment by bhavukkalra** [comment]
Survey currently under review by the community [No additions have been made, other than the installer options already listed in the original issue]. These are the installations  - 

Anaconda - https://docs.anaconda.com/free/anaconda/install/index.html
MiniConda - https://docs.conda.io/en/latest/miniconda.html
MiniForge - https://github.com/conda-forge/miniforge
mambaforge - https://github.com/conda-forge/miniforge#mambaforge
micromamba - https://mamba.readthedocs.io/en/latest/installation.html#micromamba

**Comment by bhavukkalra** [comment]
Just wanted to update on the status of this issue.
- Working on summarizing the inputs gathered from the matrix chat
- Replicating the UI experience of https://pytorch.org/get-started/locally/, with custom fields (Will share a sample here).

**Comment by bhavukkalra** [comment]
Summary - 

These are the different installation ways that are being considered for addition

[Anaconda](https://docs.anaconda.com/free/anaconda/install/index.html)
[MiniConda](https://docs.conda.io/en/latest/miniconda.html)
[MiniForge](https://github.com/conda-forge/miniforge)
[mambaforge](https://github.com/conda-forge/miniforge#mambaforge)
[micromamba](https://mamba.readthedocs.io/en/latest/installation.html#micromamba)
[condacolab](https://pypi.org/project/condacolab/) (Only for Google Colab)
[ensureconda](https://github.com/conda-incubator/ensureconda)
[brew](https://formulae.brew.sh/cask/anaconda)
nuget (Couldn't find the website for this)
[scoop](https://scoop.sh/#/apps?q=conda&s=0&d=1&o=true) (Can install mamba forge using scoop)





**Below is a sample widget**
Inspiration from - https://pytorch.org/get-started/locally/

TODO
- Rendering the sample output command based on the widgets selected above is left, but should be done soon

---
Additon to be made - "the Python versions for which the different installers are available (Anaconda and Miniconda offer several, but Miniforge and Mambaforge only one; and this info doesn't really apply to micromamba), and the operating systems / architectures."

I believe we can address this by just listing out Python version available alongside the version selected from the widget.
Or should we make a separate selection just for this?

Example - 

![image](https://user-images.githubusercontent.com/41142068/236529755-6e8f26d3-1e48-4af0-9305-12cc4dfd9622.png)


<h1>Sample Prototype</h1>

Few Questions - 

- Does the below the implementation looks good? Do let me know if there are any new tabs that could be added
- Do we have a standardized color palette for the website? So that I could make the [Tabs] look in sync with the website

![Main](https://user-images.githubusercontent.com/41142068/236525788-e9ceff50-bbbf-4a73-aa0d-82bb1e236bba.gif)

**Comment by bhavukkalra** [comment]
Here is the **Final first draft** widget Interaction

Code - https://github.com/bhavukkalra/react-samples

![UploadThis](https://user-images.githubusercontent.com/41142068/236675907-7074d198-b436-46ff-a4e7-b373dca27526.gif)

**Comment by travishathaway** [comment]
Hi @bhavukkalra,

This looks like a good first attempt. Are you able to link to a live preview anywhere?

**Comment by bhavukkalra** [comment]
Sure, Here it is - https://bhavukkalra.github.io/react-samples/

I created a separate route for this.

**Comment by travishathaway** [comment]
@bhavukkalra, 

One thing that should be addressed is that on hover the cursor needs to turn into a pointer just  like it does when you hover over links in your browser.

I also think that the colors on the left column should be muted like they are on the PyTorch selection matrix.  I don't like the stark contrast between the green and the red. The red should be a green and the green should go away.

Speaking of colors, you should also consider choosing colors that fit the theme for the website. You can find those colors here: https://conda.org/style-guide#color-palette

**Comment by jaimergp** [comment]
I think we still need the Python implementation (CPython, PyPy) and version(3.8-3.11?) selector (for some combinations only), and the architecture(s) (linux-64, linux-aarch64, linux-ppc64le, osx-64, osx-arm64, win-64...).

**Comment by jaimergp** [comment]
One more reference, the rapids.ai page:

* https://docs.rapids.ai/install#selector
* Source: https://github.com/pytorch/pytorch.github.io/blob/site/assets/quick-start-module.js

**Comment by bhavukkalra** [comment]
> One more reference, the rapids.ai page:
> 
> * https://docs.rapids.ai/install#selector
> * Source: https://github.com/pytorch/pytorch.github.io/blob/site/assets/quick-start-module.js

This is actually quiet helpful. Implementing this straight in code could be quiet a task with all the slider logic that needs to be made. 
@jaimergp What do you suggest? Should we make a final design first, like in figma or something, as that could give us a guiding direction.

As once that is finalized, I believe It won't take long to convert that into code. (As we can use the above as a template)

**Comment by jaimergp** [comment]
If it's easier for you to do it in Figma, go ahead!

**Comment by Avik-creator** [comment]
Can I contribute to this?

**Comment by tnabtaf** [comment]
Hi @Avik-creator, I think this effort stalled.  If you would be interested in working on it, then I encourage you to take it on.

**Comment by travishathaway** [comment]
@Avik-creator, I added you as co-assignee. Let us know if you need clarification or help.

**Comment by Avik-creator** [comment]
Sure sorry for the very late reply

**Comment by travishathaway** [comment]
@Avik-creator,

The issue is still available to work on.

**Comment by Avik-creator** [comment]
Ok thanks

**Comment by Avik-creator** [comment]
@travishathaway can we make it like this?

![IMG_20240902_162430.jpg](https://github.com/user-attachments/assets/04f55841-c9a6-43f2-9c5f-04f279367aee)

**Comment by travishathaway** [comment]
@Avik-creator,

No. The selection of a conda installer depends on at least two variables:

1. Operating system
2. Installer type (e.g. Miniforge, Miniconda or Anaconda)

So, it needs to reflect those.

**Comment by Avik-creator** [comment]
Ok, so the layout is not the problem right? I will add those then

**Comment by Avik-creator** [comment]
Deployed URL of the Prototype: https://guileless-salmiakki-3e2ec6.netlify.app


Github URL: https://github.com/Avik-creator/conda-forge


@travishathaway please check it once, if possible if you wanted it like this or not, then will do the integration in the main site

**Comment by travishathaway** [comment]
Thanks @Avik-creator,

This is looking good so far, but I am still not sure exactly where on the website we are going to put this.

@jaimergp, do you have any thoughts or opinions on this?

**Comment by Avik-creator** [comment]
I guess what we can do is to have a install button in the home page and then create a modal that will pop up containing this?

**Comment by travishathaway** [comment]
> I guess what we can do is to have a install button in the home page and then create a pop up modal containing this?
> 
> @travishathaway

Yes, that is one possibility. If it's not too difficult to implement, it would be really nice to see a pull request with this implemented. Each pull request generates a hosted preview of the website we can examine.

If you need more certainty before working on a such a thing, hold off for now and let me talk with some of the other project maintainers to make sure this is really something we would like to have on the website.

**Comment by Avik-creator** [comment]
Sure then will make a pr with this being implemented. You can check it there once.

**Comment by jaimergp** [comment]
We'll also need the architecture of the operating system:

- Linux x64, aarch64, ppc64le
- macOS x64, ARM64
- Windows x64, (and soon ARM64)

And it would be nice to have the selector inform about the default channel (mostly `defaults` or `conda-forge`).

Then we need a way to include micromamba and pixi in the matrix because they are self-contained tools in a way.

**Comment by jaimergp** [comment]
>  I am still not sure exactly where on the website we are going to put this.

It should be in a page where we discuss the different ways of getting started with the conda packaging ecosystem. "Choose your conda package manager" would be the section where this widget allows an easy way to choose what they user might want.

**Comment by Avik-creator** [comment]
@jaimergp @travishathaway  Something like this? The URL's might be broken for some I should say, for the installation modal.

https://guileless-salmiakki-3e2ec6.netlify.app

**Comment by jaimergp** [comment]
That's a super solid start! Love it! Do you think you can make it work with our Docusaurus site? It should be more or less straight forward with React (but I haven't seen the code behind your widget, so ymmv). Once there's a PR opened in this repo we can provide more specific feedback (e.g. wording or dropdown order).

**Comment by Avik-creator** [comment]
I think we could do it. I was using tailwind css but let me just do it using plain css and make it good enough. Could you please share the picture of the place where we should place it in the website so that I could start working on it. @travishathaway @jaimergp 

Thank you in advance. Should I place it here?
![Conda-FAQ](https://github.com/user-attachments/assets/d7a336f4-7b34-4a78-9c9c-302f01fcbfb7)

**Comment by jaimergp** [comment]
You can create a new page under Learn: `learn/installers.mdx` or something. Docusaurus uses its own framework "Infima" so there are _some_ components you can reuse (but it's not as full featured as others). Check https://docusaurus.io/docs/styling-layout#global-styles.

**Comment by Avik-creator** [comment]
@travishathaway @jaimergp Created the PR. Please check it once and do let me know If I need to change somethings.
